packages/next/src/build/swc/index.ts

```diff
@@ -1130,12 +1133,61 @@ function bindingToApi(
     }
   }
 
+  /**
+   * Returns a new copy of next.js config object to avoid mutating the original.
+   *
+   * Also it does some augmentation to the configuration as well, for example set the
+   * turbopack's rules if `experimental.reactCompilerOptions` is set.
+   */
+  function augmentNextConfig(
+    nextConfig: NextConfigComplete,
+    projectPath: string
+  ): Record<string, any> {
+    let ret = { ...(nextConfig as any) }
+
+    const reactCompilerOptions = ret.experimental?.reactCompiler
+
+    // It is not easy to set the rules inside of rust as resolving, and passing the context identical to the webpack
+    // config is bit hard, also we can reuse same codes between webpack config in here.
+    if (reactCompilerOptions) {
+      if (!ret.experimental.turbo) {
+        ret.experimental.turbo = {}
+      }
+
+      if (!ret.experimental.turbo.rules) {
+        ret.experimental.turbo.rules = {}
+      }
+
+      // [TODO]: need to enable against '*.ts'
+      for (const key of ['*.ts', '*.js', '*.jsx', '*.tsx']) {
+        if (ret.experimental.turbo.rules[key]) {
+          Log.warn(
+            `Next config have a rule for the ${key} for Turbopack, not able to enable react compiler automatically. Manually add babel-loader's plugin insetad.`
```

Comment:
I think something like this would be more helpful?
```suggestion
            `The React Compiler cannot be enabled automatically because `experimental.turbo` contains a rule for '${key}'. Remove this rule, or add 'babel-loader' and 'babel-plugin-react-compiler' to the turbo configuration manually.`
```
It'd be good to explain the process of adding it manually. Also, will the rest of this setup work at all if it's not added here?

Reply:
If this hits, we don't setup anything for the react compiler and user need to configure everything manually.